app/controllers/notes_controller.rb

```diff
@@ -55,10 +55,56 @@ def destroy
       track_destroyed(@note)
       redirect_to node_path(@node), notice: 'Note deleted'
     else
-      redirect_to node_note_path(@node, @note), alert: 'Could not delete node'
+      redirect_to node_note_path(@node, @note), alert: 'Could not delete note'
     end
   end
 
+  def multi_destroy
+    @notes = Note.where(
+      'node_id = :node AND id in (:ids)',
+      node: @node,
+      ids: params[:ids]
+    )
+
+    if @notes.any?
+      @job_logger = Log.new(uid: (Log.maximum(:uid) || 0) + 1)
+
+      if @notes.count > Note::MAX_DELETED_INLINE
+        @job_logger.write 'Enqueueing destroy job to start in the background.'
+        job = DestroyJob.perform_later(
+          items: @notes.to_a,
```

Comment:
I believe that passing a class name (as string) and the list of ids instead of the full AR objects was the recommended way to handle data exchange between the app and background workers. Does it ring a bell?